ui/components/app/snaps/snap-install-warning/snap-install-warning.js

```diff
@@ -29,118 +41,117 @@ const checkboxStateReducer = produce((state, action) => {
       throw new Error(
         'You must provide a type when dispatching an action for checkboxState',
       );
   }
 });
-
 export default function SnapInstallWarning({
   onCancel,
   onSubmit,
   warnings,
   snapName,
+  isOpen,
 }) {
   const t = useI18nContext();
   const [checkboxState, dispatch] = useReducer(checkboxStateReducer, {});
-
   const isAllChecked = warnings.every((warning) => checkboxState[warning.id]);
-
   const onCheckboxClicked = useCallback((checkboxId) => {
     dispatch({ type: 'check', checkboxId });
   }, []);
 
-  const SnapInstallWarningFooter = () => {
-    return (
-      <div className="snap-install-warning__footer">
+  return (
+    <Modal className="snap-install-warning" onClose={onCancel} isOpen={isOpen}>
+      <ModalOverlay />
+      <ModalContent
+        modalDialogProps={{
+          display: Display.Flex,
+          flexDirection: FlexDirection.Column,
+          gap: 4,
+        }}
+      >
+        <ModalHeader
+          onClose={onCancel}
+          childrenWrapperProps={{
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+            alignItems: AlignItems.center,
+            justifyContent: JustifyContent.Center,
+            gap: 4,
+          }}
+        >
+          <AvatarIcon
+            iconName={IconName.Danger}
+            backgroundColor={BackgroundColor.warningMuted}
+            color={IconColor.warningDefault}
+            size={Size.XL}
+          />
+          <Text variant={TextVariant.headingMd} textAlign={TextAlign.Center}>
+            {t('snapInstallWarningHeading')}
+          </Text>
+        </ModalHeader>
+        <Text textAlign={TextAlign.Center} marginBottom={4}>
+          {warnings.length > 1
+            ? t('snapInstallWarningCheckPlural', [
+                <Text
+                  key="snapNameInWarningDescription"
+                  fontWeight={FontWeight.Medium}
+                  as="span"
+                >
+                  {snapName}
+                </Text>,
+              ])
+            : t('snapInstallWarningCheck', [
+                <Text
+                  key="snapNameInWarningDescription"
+                  fontWeight={FontWeight.Medium}
+                  as="span"
+                >
+                  {snapName}
+                </Text>,
+              ])}
+        </Text>
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          gap={4}
+          marginBottom={4}
+        >
+          {warnings.map((warning) => (
+            <Box
+              key={warning.id}
+              display={Display.Flex}
+              gap={2}
+              alignItems={AlignItems.center}
+            >
+              <CheckBox
+                checked={checkboxState[warning.id] ?? false}
+                id={warning.id}
+                onClick={() => onCheckboxClicked(warning.id)}
+              />
+              <Label htmlFor={warning.id}>{warning.message}</Label>
```

Comment:
we recently merged a new checkbox component so you could update this as well to use the one from the component-library. The component props are different so make sure to read the docs https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-checkbox--docs

Reply:
updated screenshots and made changes [`here`](https://github.com/MetaMask/metamask-extension/pull/20284/commits/a16289568d1dd752815167fadfb16bb5a0abdedd)